animations/lib/src/misc/curved_animation.dart

```diff
@@ -0,0 +1,146 @@
+// Copyright 2019 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'dart:math' as math;
+
+class CurvedAnimationDemo extends StatefulWidget {
+  static const String routeName = '/misc/curved_animation';
+  @override
+  _CurvedAnimationDemoState createState() => _CurvedAnimationDemoState();
+}
+
+class MyCurve {
```

Comment:
Rather than `MyCurve`, consider `CurveChoice` or `CurveSetting` as a name, since `MyCurve` is a fairly generic name.